File: exercise/java/week01-code-analysis/exercise/src/main/java/movie/StoreAccount.java
package movie;

import java.util.ArrayList;
import java.util.List;

public class StoreAccount {
    public Double totalSold = 0d;

    public List<MovieSale> allSales;

    public StoreAccount() {
        this.allSales = new ArrayList<>();
    }

    public void sell(Movie movie, String to) {
        totalSold += movie.unitPrice;
        allSales.add(new MovieSale(to, movie.title));
    }

    static class MovieSale {
        String clientName;
        String movieName;

        public MovieSale(String clientName, String movieName) {
            this.clientName = clientName;
            this.movieName = movieName;
        }
    }
}

File: exercise/c#/week01-code-analysis/ExerciseMovieStore/StoreAccount.cs
namespace ExerciseMovieStore;

public class StoreAccount
{
    public double TotalSold { get; set; } = 0d;
    public List<MovieSale> AllSales { get; set; }

    public StoreAccount()
    {
        AllSales = new List<MovieSale>();
    }

    public void Sell(Movie movie, string to)
    {
        TotalSold += movie.UnitPrice;
        AllSales.Add(new MovieSale(to, movie.Title));
    }

    public class MovieSale
    {
        public string ClientName { get; set; }
        public string MovieName { get; set; }

        public MovieSale(string clientName, string movieName)
        {
            ClientName = clientName;
            MovieName = movieName;
        }
    }
}